### PR TITLE
[Autocomplete]: emit select of textbox value when no selection is made.

### PR DESCRIPTION
**Pull request by ryatziv**
Modifications:
1. Pressing up in the highest suggestion selection will unselect.
2. Clear selected index after a selection
3. Pressing enter when there is no selection will emit the current value as a selection

I am happy to put these behind a prop flag, like `select-without-value`.

Please make sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

**Comment by Leopoldthecoder** [comment]
LGTM. Like you said, I think it's better to add a prop. Maybe `select-when-unmatched`?

**Comment by ryatziv** [comment]
Cool, I will add the prop

**Comment by Leopoldthecoder** [comment]
I think just for #3 would be fine. What do you think?

File: examples/docs/en-US/input.md [hunk 1]
```diff
@@ -596,26 +596,26 @@ Search data from server-side.
...
-|value| binding value | string/number| — | — |
-|maxlength| maximum Input text length| number| — | — |
-|minlength| minimum Input text length| number | — | — |
-|placeholder| placeholder of Input| string | — | — |
-|disabled | whether Input is disabled | boolean | — | false |
-|size | size of Input, works when `type` is not 'textarea' | string | large/small/mini | — |
-|icon | icon name | string | — | — |
-|rows | number of rows of textarea, only works when `type` is 'textarea' | number | — | 2 |
-|autosize | whether textarea has an adaptive height, only works when `type` is 'textarea'. Can accept an object, e.g. { minRows: 2, maxRows: 6 }  | boolean/object | — | false |
+|type| Same as the `type` attribute of native input, except that it can be `textarea` | string | - | text |
```
Comment (ryatziv) [inline examples/docs/en-US/input.md]: I'm having trouble with the dashes.  When I updated from dev, everything came in as `???`.  When I changed them to dashes, they don't look like the dashes that you are using.  I tried to copy/paste from another MD file, but when I run the build, it renders as `?`.  Can you help me out?

Comment (Leopoldthecoder) [reply examples/docs/en-US/input.md]: I'll take care of it.

**Comment by ryatziv** [comment]
Thank you :)

**Pull request merged**